scripts/train.py

```diff
@@ -0,0 +1,181 @@
+
+# TODO(diwei): remove
+
+
+def count_parameters(model):
+    return sum(p.numel() for p in model.parameters() if p.requires_grad)
+
+
+# TODO(diwei): remove
+def initialize_weights(m):
+    if hasattr(m, 'weight') and m.weight.dim() > 1:
+        nn.init.kaiming_uniform(m.weight.data)
+
+
+# TODO(diwei): remove
+def epoch_time(start_time, end_time):
+    elapsed_time = end_time - start_time
+    elapsed_mins = int(elapsed_time / 60)
+    elapsed_secs = int(elapsed_time - (elapsed_mins * 60))
+    return elapsed_mins, elapsed_secs
+
+
+# TODO(diwei): remove
+print(f'The model has {count_parameters(model):,} trainable parameters')
+model.apply(initialize_weights)
+
+
+# TODO(diwei): adjust
+optimizer = Adam(params=model.parameters(),
+                 lr=top_config["opt_params"]["lr"],
+                 weight_decay=top_config["opt_params"]["weight_decay"],
+                 eps=top_config["opt_params"]["eps"])
+
+scheduler = optim.lr_scheduler.ReduceLROnPlateau(optimizer=optimizer,
+                                                 verbose=True)
+
+# TODO(diwei): adjust
+criterion = nn.CrossEntropyLoss()
+
+
+def train(model, loader, optimizer, criterion, device='cpu'):
+    """THe interface to train the model using training data
+
+    Parameters
+    ----------
+    model : nn.Module
+        The TopFormer model used to train.
+    loader : DataLoader
+        The loader to visit the data.
+    optimizer : Adam
+        Optimizer to iterate the params.
+    criterion : Loss nn.CrossEntropyLoss
+        Loss criterion.
+    device : str, optional
+        Device used to evaluate, by default 'cpu'
+
+    Returns
+    -------
+    float
+        Average loss.
+    """
+    epoch_loss = 0
+    model.to(device)
+    for i, batch in enumerate(loader):
+        x = batch["x"]
+        u = batch["u"]
+        gt = batch["gt"]
+
+        optimizer.zero_grad()
+        output = model(x.permute(1, 0, 2), u.permute(1, 0, 2))
+        loss = criterion(torch.max(output, dim=0)[0], gt)
+        loss.backward()
+        optimizer.step()
+
+        epoch_loss += loss.item()
+
+        # TODO(diwei): use other progressbar
+        print('step :', round((i / (len(loader) + + sys.float_info.epsilon)) * 100, 2),
+              '% , loss :', loss.item())
+
+    return epoch_loss / (len(loader) + sys.float_info.epsilon)
+
+
+def evaluate(model, loader, criterion, device='cpu'):
+    """The interface to evaluate the model using validation data
+
+    Parameters
+    ----------
+    model : nn.Module
+        The TopFormer model used to evaluate.
+    loader : DataLoader
+        The loader to visit the data.
+    criterion : Loss nn.CrossEntropyLoss
+        Loss criterion.
+    device : str, optional
+        Device used to evaluate, by default 'cpu'.
+
+    Returns
+    -------
+    float
+        Average loss.
+    """
+    epoch_loss = 0
+    model.to(device)
```

Comment:
model.eval() for managing evaluation mode

Reply:
Done.Good advice.